Add missing exceptions documentation related to storage_backend

MVrachev opened the pull request:
### Description of the changes being introduced by the pull request:

The missing exceptions documentation that was added is related to the
code for the "storage_backend" and its API calls.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

adityasaky commented on securesystemslib/hash.py:
```diff
@@ -343,6 +343,8 @@ def digest_filename(filename, algorithm=DEFAULT_HASH_ALGORITHM,
     securesystemslib.exceptions.UnsupportedLibraryError, if the given
     'hash_library' is unsupported.
 
+    securesystemslib.exceptions.StorageError, if the file cannot be open.
```
```suggestion
    securesystemslib.exceptions.StorageError, if the file cannot be opened.
```

adityasaky commented on securesystemslib/util.py:
```diff
@@ -109,6 +109,9 @@ def get_file_hashes(
     securesystemslib.exceptions.Error: If 'filepath' does not exist.
 
+    securesystemslib.exceptions.StorageError: The file at "filepath" cannot be
+    open or found.
```
```suggestion
    opened or found.
```

adityasaky reviewed (approved):
Nits, but LGTM!

MVrachev commented:
> Nits, but LGTM!

I updated the pr addressing your comments and updated the exceptions in the test.
Plus added a small use case that wasn't tested.

MVrachev commented:
PS: The error happening on build https://github.com/secure-systems-lab/securesystemslib/runs/4884928637?check_suite_focus=true is not related to my changes.

adityasaky commented:
Re-run seems to have fixed it :)
